### PR TITLE
Template fixes

### PR DESCRIPTION
**Pull request by certik**

### CODE REVIEW

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -2462,7 +2462,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
             } else {
                 if (!v) {
                     if (is_template) {
-                      throw SemanticError("Type parameter '" + derived_type_name + "' not required", loc);
+                      throw SemanticError("Type parameter '" + derived_type_name + "' is not specified " 
+                                          "in any requirements", loc);
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:2466]:
```suggestion
                                          "in any requirement", loc);
```

**Comment by certik** [comment]
@ansharlubis what is the status of this -- do you want to rebase it on top of the latest master?

**Comment by ansharlubis** [comment]
@certik I still have a few more small fixes that I want to add to this PR.

**Comment by ansharlubis** [comment]
@certik I got all the fixes I need to add for now. There will be another PR for synchronizing template instantiation with LPython.

**Pull request merged**